File: core/src/commonMain/kotlin/com/jerreader/unified/domain/LanguageCode.kt
package com.jerreader.unified.domain

enum class LanguageCode(val tag: String) {
    CHINESE_SIMPLIFIED("zh-Hans"),
    ENGLISH("en"),
    JAPANESE("ja");

    companion object {
        fun fromTag(tag: String): LanguageCode? =
            entries.firstOrNull { language -> language.tag.equals(tag, ignoreCase = true) }
    }
}

File: iosApp/Jerreader/Core/Models/LanguageCode.swift
import Foundation

enum LanguageCode: String, CaseIterable, Codable, Hashable, Sendable {
    case japanese = "ja"
    case english = "en"
    case simplifiedChinese = "zh-Hans"

    var displayName: String {
        switch self {
        case .japanese:
            return "日语"
        case .english:
            return "英语"
        case .simplifiedChinese:
            return "简体中文"
        }
    }
}
